src/components/Garden.js

```diff
@@ -0,0 +1,143 @@
+  petalGroup.append('path')
+    .attr('d', petalPathThree)
+    .attr('fill', d => colorsByLanguage[d.languages[2]] || 'none')
+    .attr('transform', `translate(-50, -180)`)
+  petalGroup.append('path')
+    .attr('d', petalPathThree)
+    .attr('fill', d => colorsByLanguage[d.languages[2]] || 'none')
+    .attr('transform', `translate(-50, -180)rotate(120, 50, 160)`)
+  petalGroup.append('path')
+    .attr('d', petalPathThree)
+    .attr('fill', d => colorsByLanguage[d.languages[2]] || 'none')
+    .attr('transform', `translate(-50, -180)rotate(240 50 160)`)
+  petalGroup.append('path')
+    .attr('d', petalPathTwo)
+    .attr('fill', d => colorsByLanguage[d.languages[1]] || 'none')
+    .attr('transform', `translate(-50, -150)rotate(60 50 130)`)
+  petalGroup.append('path')
+    .attr('d', petalPathTwo)
+    .attr('fill', d => colorsByLanguage[d.languages[1]] || 'none')
+    .attr('transform', `translate(-50, -150)rotate(180 50 130)`)
+  petalGroup.append('path')
+    .attr('d', petalPathTwo)
+    .attr('fill', d => colorsByLanguage[d.languages[1]] || 'none')
+    .attr('transform', `translate(-50, -150)rotate(300 50 130)`)
+  petalGroup.append('path')
+    .attr('d', petalPathOne)
+    .attr('fill', d => colorsByLanguage[d.languages[0]] || 'none')
+    .attr('transform', `translate(-50, -130)`)
+  petalGroup.append('path')
+    .attr('d', petalPathOne)
+    .attr('fill', d => colorsByLanguage[d.languages[0]] || 'none')
+    .attr('transform', `translate(-50, -130)rotate(120, 50, 110)`)
+  petalGroup.append('path')
+    .attr('d', petalPathOne)
+    .attr('fill', d => colorsByLanguage[d.languages[0]] || 'none')
+    .attr('transform', `translate(-50, -130)rotate(240,  50, 110)`)
+
+  //   DRAW STEMS
+  //   height scaled to activeLife
+    const stem = flowerBed.selectAll('.stem')
+      .data(repositories).enter()
+      .append('path')
+      .attr('class','stem')
+      .attr('id', (d, i) => `myStem${i}`)
+      .attr('d', d => `M0,600 C 80 ${yStemScale(d.lifespan)}, -20 ${yStemScale(d.lifespan)}, 0 ${yStemScale(d.lifespan)}`)
+      .attr('transform', (d, i) => `translate(${100 + i * 200},0)`)
+      .attr('stroke-width', 5)
+      .attr('stroke', 'green')
+      .attr('fill', 'none')
+
+      const animate = () => {
+        flowerBed.selectAll('.stem')
+          .transition().duration(2000)
+          .attr('d', d => `M0,600 C -20 ${yStemScale(d.lifespan)}, 20 ${yStemScale(d.lifespan)}, 0 ${yStemScale(d.lifespan)}`)
+            .attr('stroke', 'green')
+          .transition().duration(2000)
+          .attr('d', d => `M0,600 C 80 ${yStemScale(d.lifespan)}, -20 ${yStemScale(d.lifespan)}, 0 ${yStemScale(d.lifespan)}`)
+          .on("end", animate)
+      }
+    animate();
+
+
+    flowerBed.selectAll('.repoName')
+      .data(repositories).enter()
+      .append('text')
+      .append('textPath')
+      .attr('xlink:href', (d, i) => `#myStem${i}`)
```

Comment:
ah I see, so text is assigned to path with corresponding index, which you had to assign dynamically